examples/md-flexible/src/Simulation.h

```diff
@@ -512,3 +521,55 @@ void Simulation<Particle, ParticleCell>::writeVTKFile(unsigned int iteration,
 
   _timers.vtk.stop();
 }
+
+template <class Particle, class ParticleCell>
+double Simulation<Particle, ParticleCell>::getHomogeneity(autopas::AutoPas<Particle, ParticleCell> &autopas) {
+
+  std::array<double, 3> startCorner = autopas.getBoxMin();
+  std::array<double, 3> endCorner = autopas.getBoxMax();
+  std::array<int, 3> numberOfCells = {};
+  int cellSize = 3;
+
+  for(int i = 0; i < 3; ++i){
+    numberOfCells[i] = ceil((endCorner[i] - startCorner[i]) / cellSize);
+  }
+
+  int totalNumberOfCells = numberOfCells[0] * numberOfCells[1] * numberOfCells[2];
+
+  std::vector<int> particlesPerCell = {};
+  std::vector<double> densityPerCell = {};
+
+  std::array<double, 3> lowCorner = {};
+  std::array<double, 3> highCorner = {};
+
+  for(int i = 0; i < numberOfCells[0]; ++i){
+    for(int j = 0; j < numberOfCells[1]; ++j){
+      for(int k = 0; k < numberOfCells[2]; ++k){
+        double cellSizeDouble = double (cellSize);
+        double x = double (i*cellSize);
+        double y = double (j*cellSize);
+        double z = double (k*cellSize);
+        lowCorner = {x, y, z};
+        highCorner = {x + cellSizeDouble, y + cellSizeDouble, z + cellSizeDouble};
+        int counter = 0;
+        for (auto iter = autopas.getRegionIterator(lowCorner, highCorner); iter.isValid(); ++iter) {
+          counter++;
+        }
+        double size = pow(cellSizeDouble, 3);
+        particlesPerCell.push_back(counter);
+        densityPerCell.push_back(counter / size);
+      }
+    }
+  }
+
+  double mean = autopas.getNumberOfParticles() / (endCorner[0] * endCorner[1] * endCorner[2]);
+  double variance = 0.0;
+
+  for(int r = 0; r < densityPerCell.size(); ++r){
+    variance += (pow((densityPerCell.at(r) - mean), 2.0) / densityPerCell.size());
```

Comment:
```suggestion
    variance += (densityPerCell.at(r) - mean)*(densityPerCell.at(r) - mean) / densityPerCell.size();
```

Reply:
using 2.0 is mostly ok, because it is almost always optimized out, but still not so good.

Reply:
Actually, in this case, I would prefer the `pow` version since it is more readable.
For the case that the exponent is 2 `pow` and the multiplication have the same performance (as you can see [here](https://quick-bench.com/q/8Tak-Luy597pDB2AtvUeXt5zjds)). Whenever the exponent is not 2 the difference in speed is significant. double or int does not matter in either case.

Reply:
So do I leave the pow-version as it is, or shall I change it?

Reply:
as you like. pow might be more readable. alternatively, you can introduce an intermediate variable `distance` or so and then use `distance*distance`.